token/core/common/interop/pledge/validator_pledge.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package pledge
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/hyperledger-labs/fabric-token-sdk/token/core/common"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/driver"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/identity"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/interop/pledge"
+	"github.com/pkg/errors"
+)
+
+func IssuePledgeValidate[P driver.PublicParameters, T driver.Output, TA driver.TransferAction, IA driver.IssueAction, DS driver.Deserializer](ctx *common.Context[P, T, TA, IA, DS]) error {
+	for k := range ctx.IssueAction.GetMetadata() {
+		ctx.CountMetadataKey(k)
+	}
+	return nil
+}
+
+func TransferPledgeValidate[P driver.PublicParameters, T driver.Output, TA driver.TransferAction, IA driver.IssueAction, DS driver.Deserializer](ctx *common.Context[P, T, TA, IA, DS]) error {
+	for _, in := range ctx.InputTokens {
+		id, err := identity.UnmarshalTypedIdentity(in.GetOwner())
+		if err != nil {
+			return errors.Wrap(err, "failed to unmarshal owner of input token")
+		}
+		if id.Type == pledge.ScriptType {
+			if len(ctx.InputTokens) != 1 || len(ctx.TransferAction.GetOutputs()) != 1 {
+				return errors.Errorf("invalid transfer action: a pledge script only transfers the ownership of a token")
+			}
+			out := ctx.TransferAction.GetOutputs()[0]
+			sender, err := identity.UnmarshalTypedIdentity(ctx.InputTokens[0].GetOwner())
+			if err != nil {
+				return err
+			}
+			script := &pledge.Script{}
+			err = json.Unmarshal(sender.Identity, script)
+			if err != nil {
+				return err
+			}
+			if time.Now().Before(script.Deadline) {
+				return errors.New("cannot reclaim pledge yet: wait for timeout to elapse.")
```

Comment:
It's bot reclaim and redeem. Shall we say `cannot change the state of the pledge yet`?